### PR TITLE
feat: composable for tiered badge data

### PR DESCRIPTION
**Pull request by dyersituations**
https://kiva.atlassian.net/browse/MP-966

- Partial implementation to allow for other engineers to utilize this work
- Created composable for combining badge tiered achievements and contentful data
- Work to come: comments, unit tests, using on MyKiva correctly, etc.
- This doesn't break the page more than it already is currently 😄

### CODE REVIEW

File: src/composables/useBadgeData.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+	 *   "level": undefined
+	 * }
+	 */
+	const badgeData = computed(() => {
+		const badges = [];
+
+		// Ensure data loaded from both achievement service and Contentful
+		if (badgeAchievementData.value && badgeContentfulData.value) {
+			// Currently only targeting specific tiered badges
+			defaultBadges.forEach(badgeKey => {
```
Comment (roger-in-kiva) [inline src/composables/useBadgeData.js:84]: I guess we need a more generic data return and not for default ones only, as in my achievements section we need to verify all of them and show the earned tiers as well.

Comment (dyersituations) [reply src/composables/useBadgeData.js:84]: Ah, I didn't realize we were showing everything in that section. Okay, I'll change this real quick.

Comment (dyersituations) [reply src/composables/useBadgeData.js]: Done.

**Review by roger-in-kiva — APPROVED** [review]
Looks good, thanks!

**Pull request merged**